src/datasources/email/email.datasource.ts

```diff
@@ -0,0 +1,104 @@
+import { Inject, Injectable } from '@nestjs/common';
+import * as postgres from 'postgres';
+import { IEmailDataSource } from '@/domain/interfaces/email.datasource.interface';
+import { EmailVerificationCode } from '@/domain/email/entities/email-verification-code.entity';
+import { EmailAddressDoesNotExistError } from '@/datasources/email/errors/email-address-does-not-exist.error';
+import { Email } from '@/datasources/email/entities/email.entity';
+
+@Injectable()
+export class EmailDataSource implements IEmailDataSource {
+  constructor(@Inject('DB_INSTANCE') private readonly sql: postgres.Sql) {}
+
+  private async _getEmail(args: {
+    chainId: string;
+    safeAddress: string;
+    signer: string;
+  }): Promise<Email> {
+    const [email] = await this.sql<Email[]>`SELECT *
+                                            FROM emails.signer_emails
+                                            WHERE chain_id = ${args.chainId}
+                                              and safe_address = ${args.safeAddress}
+                                              and signer = ${args.signer}`;
+
+    if (!email) {
+      throw new EmailAddressDoesNotExistError(
+        args.chainId,
+        args.safeAddress,
+        args.signer,
+      );
+    }
+
+    return email;
+  }
```

Comment:
This function is not used anymore after the refactor, right?

Reply:
Yes it's not longer required! Removed in [641cb20](https://github.com/safe-global/safe-client-gateway/pull/829/commits/641cb204025e4ca4b2373dd5904ad99710b7da68)